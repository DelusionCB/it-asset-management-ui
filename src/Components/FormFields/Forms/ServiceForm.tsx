import React, {useEffect, useState} from 'react';
import {Row} from 'reactstrap';
import {
    serviceDataProps,
} from '../../../Types/types.directories';
import {serviceDefaultValues} from '../../../Types/defaultValues';
import TextField from '../../CustomComponents/TextField';
import SearchableSelect from '../../CustomComponents/SearchableSelect';
import {
    debouncedCheck,
    handleClear,
    handleServiceChange, handleServiceSelect,
} from '../../../Utils/helpers';
import {useTranslation} from 'react-i18next';
import ActionButtons from '../../ActionButtons/ActionButtons';
import VisibilitySelector from '../../CustomComponents/VisibilitySelector';
import {formTypes} from '../../../Types/types.forms';
import StaticSelect from '../../CustomComponents/StaticSelect';
import AsyncSelect from '../../CustomComponents/AsyncSelect';

function ServiceForm ({isDisabled, type, mode, editValues}: formTypes): JSX.Element {
    const [values, setValues] = useState<serviceDataProps>(serviceDefaultValues)
    const {t} = useTranslation()

    function handleChange (e: React.ChangeEvent<HTMLInputElement>): void {
        handleServiceChange(e, values, setValues);
    }

    function handleSelect (value: {} | [] | undefined | null, id: string): void {
        handleServiceSelect(value, id, values, setValues);
    }

    useEffect(() => {
        if (Object.keys(editValues).length !== 0 && (mode === 'edit' || mode === 'copy')) {
            setValues(editValues)
        }
        if (Object.keys(editValues).length !== 0 && mode === 'create') {
            setValues(serviceDefaultValues)
        }
    }, [editValues, mode])

    useEffect(() => {
        if (mode !== 'edit') {
            // Invoke debouncedCheck with updated values
            debouncedCheck(values, serviceDefaultValues, isDisabled);

            // Cleanup function to cancel debounce on unmount
            return () => {
                debouncedCheck.cancel();
            };
        }
    }, [values]);

    useEffect(() => {
        if (values.visibility === 'hidden' || values.visibility === 'draft') {
            handleClear<serviceDataProps>(
                values,
                ['contract', 'provider'],
                ['related_services', 'required_installations'],
                setValues,
            )
        }
    }, [values.visibility])

    return (
        <>
            <Row>
                <h1>{t(`service.${mode}`)}</h1>
            </Row>
            <Row>
                <h2>{t('service.visibility')}</h2>
            </Row>
            <Row>
                <VisibilitySelector
                    value={values.visibility}
                    onChange={(e) => { handleChange(e); }}
                    disabled={false}
                    id='visibility'
                    label={'service.values.visibility'}
                    invalid={false}
                    options={['draft', 'hidden', 'published']}
                    placeholder={'placeholder.visibility'}
                />
            </Row>
            <Row>
                <h2>{t('service.description')}</h2>
            </Row>
            <Row>
                <TextField
                    id='name'
                    disabled={false}
                    invalid={false}
                    label={'service.values.name'}
                    onChange={(e) => { handleChange(e); }}
                    placeholder={'placeholder.name'}
                    value={values.name}
                    type='text'
                />
            </Row>
            <Row>
                <StaticSelect
                    id={'service_status'}
                    disabled={false}
                    invalid={false}
                    label={'service.values.service_status'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.service_status'}
                    value={values.service_status}
                    options={['in_deployment', 'in_use', 'unused', 'shutdown', 'archived', 'undefined', 'unreleased', 'deleted']}
                />
            </Row>
            <Row>
                <TextField
                    id={'description'}
                    disabled={false}
                    invalid={false}
                    type='textarea'
                    label={'service.values.description'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.description'}
                    value={values.description}
                />
            </Row>
            <Row>
                <SearchableSelect
                    label={'service.values.contract'}
                    endpoint='contract'
                    id={'contract'}
                    placeholder={'placeholder.contract'}
                    disabled={values.visibility === 'hidden' || values.visibility === 'draft'}
                    invalid={false}
                    isMulti={false}
                    value={values.contract}
                    onChange={(value, id) => { handleSelect(value, id); }}
                />
            </Row>
            <Row>
                <StaticSelect
                    id={'criticality'}
                    disabled={false}
                    invalid={false}
                    label={'service.values.criticality'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.criticality'}
                    value={values.criticality}
                    options={['critical', 'high', 'normal', 'low', 'noclass']}
                />
            </Row>
            <Row>
                <TextField
                    id={'service_level'}
                    disabled={false}
                    invalid={false}
                    type='text'
                    label={'service.values.service_level'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.service_level'}
                    value={values.service_level}
                />
            </Row>
            <Row>
                <StaticSelect
                    id={'service_type'}
                    disabled={false}
                    invalid={false}
                    label={'service.values.service_type'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.service_type'}
                    value={values.service_type}
                    options={['onpremises', 'iaas', 'paas', 'caas', 'faas', 'saas', 'other']}
                />
            </Row>
            <Row>
                <TextField
                    id={'limitations'}
                    disabled={false}
                    invalid={false}
                    type='textarea'
                    label={'service.values.limitations'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.limitations'}
                    value={values.limitations}
                />
            </Row>
            <Row>
                <SearchableSelect
                    label={'service.values.related_services'}
                    endpoint='service'
                    id={'related_services'}
                    placeholder={'placeholder.related_services'}
                    disabled={values.visibility === 'hidden' || values.visibility === 'draft'}
                    invalid={false}
                    isMulti={true}
                    value={values.related_services}
                    onChange={(value, id) => { handleSelect(value, id); }}
                />
            </Row>
            <Row>
                <SearchableSelect
                    label={'service.values.required_installations'}
                    endpoint='application'
                    id={'required_installations'}
                    placeholder={'placeholder.required_installations'}
                    disabled={values.visibility === 'hidden' || values.visibility === 'draft'}
                    invalid={false}
                    isMulti={true}
                    value={values.required_installations}
                    onChange={(value, id) => { handleSelect(value, id); }}
                />
            </Row>
            <Row>
                <h2>{t('service.responsibility')}</h2>
            </Row>
            <Row>
                <TextField
                    id={'product_owner'}
                    disabled={false}
                    invalid={false}
                    type='text'
                    label={'service.values.product_owner'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.product_owner'}
                    value={values.product_owner}
                />
            </Row>
            <Row>
                <TextField
                    id={'service_holder'}
                    disabled={false}
                    invalid={false}
                    type='text'
                    label={'service.values.service_holder'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.service_holder'}
                    value={values.service_holder}
                />
            </Row>
            <Row>
                <SearchableSelect
                    label={'service.values.provider'}
                    endpoint='provider'
                    id={'provider'}
                    placeholder={'placeholder.provider'}
                    disabled={values.visibility === 'hidden' || values.visibility === 'draft'}
                    invalid={false}
                    isMulti={false}
                    value={values.provider}
                    onChange={(value, id) => { handleSelect(value, id); }}
                />
            </Row>
            <Row>
                <TextField
                    id={'provider_role'}
                    disabled={false}
                    invalid={false}
                    type='text'
                    label={'service.values.provider_role'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.provider_role'}
                    value={values.provider_role}
                />
            </Row>
            <Row>
                <TextField
                    id={'provider_contact'}
                    disabled={false}
                    invalid={false}
                    type='text'
                    label={'service.values.provider_contact'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.provider_contact'}
                    value={values.provider_contact}
                />
            </Row>
            <Row>
                <TextField
                    id={'additional_contacts'}
                    disabled={false}
                    invalid={false}
                    type='textarea'
                    label={'service.values.additional_contacts'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.additional_contacts'}
                    value={values.additional_contacts}
                />
            </Row>
            <Row>
                <h2>{t('service.extra')}</h2>
            </Row>
            <Row>
                <AsyncSelect
                    label={'service.values.customership'}
                    endpoint='customership'
                    id={'customership'}
                    placeholder={'placeholder.customership'}
                    disabled={false}
                    invalid={false}
                    isMulti={true}
                    isSearchable={false}
                    value={values.customership}
                    onChange={(value, id) => { handleSelect(value, id); }}
                />
            </Row>
            <Row>
                <StaticSelect
                    id={'validity_type'}
                    disabled={false}
                    invalid={false}
                    label={'service.values.validity_type'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.validity_type'}
                    value={values.validity_type}
                    options={['ongoing', 'temporary']}
                />
            </Row>
            <ActionButtons
                values={values}
                type={type}
                mode={mode}
            />
        </>
    );
}

export default ServiceForm;
