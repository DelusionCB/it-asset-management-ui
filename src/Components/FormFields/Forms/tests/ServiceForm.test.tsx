import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom'
import '@testing-library/jest-dom'
import ServiceForm from '../ServiceForm';

describe('<ServiceForm />', () => {
    let renderer: TestRenderer.ReactTestRenderer;

    // Render the component before tests
    beforeAll(() => {
        renderer = TestRenderer.create(
            <BrowserRouter>
                <ServiceForm editValues={{}} mode='create' type='service' isDisabled={jest.fn()} />
            </BrowserRouter>,
        );
    });
    describe('Snapshot', () => {
        it('renders correctly', () => {
            expect(renderer.toJSON()).toMatchSnapshot();
        })
    })
})
