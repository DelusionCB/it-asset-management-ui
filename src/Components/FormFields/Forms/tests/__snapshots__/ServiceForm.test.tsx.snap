// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServiceForm /> Snapshot renders correctly 1`] = `
[
  <div
    className="row"
  >
    <h1>
      service.create
    </h1>
  </div>,
  <div
    className="row"
  >
    <h2>
      service.visibility
    </h2>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="visibility"
        >
          values.visibility
        </label>
        <select
          aria-invalid={false}
          className="form-select"
          disabled={false}
          id="visibility"
          name="visibility"
          onChange={[Function]}
          placeholder="placeholder.visibility"
          value="published"
        >
          <option
            value="draft"
          >
            visibility.draft
          </option>
          <option
            value="hidden"
          >
            visibility.hidden
          </option>
          <option
            value="published"
          >
            visibility.published
          </option>
        </select>
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <h2>
      service.description
    </h2>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="name"
        >
          values.name
        </label>
        <input
          aria-invalid={false}
          className="form-control"
          disabled={false}
          id="name"
          name="name"
          onChange={[Function]}
          placeholder="placeholder.name"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="service_status"
        >
          values.service_status
        </label>
        <select
          aria-invalid={false}
          className="form-select"
          disabled={false}
          id="service_status"
          name="service_status"
          onChange={[Function]}
          placeholder="placeholder.service_status"
          value="undefined"
        >
          <option
            value="in_deployment"
          >
            select.in_deployment
          </option>
          <option
            value="in_use"
          >
            select.in_use
          </option>
          <option
            value="unused"
          >
            select.unused
          </option>
          <option
            value="shutdown"
          >
            select.shutdown
          </option>
          <option
            value="archived"
          >
            select.archived
          </option>
          <option
            value="undefined"
          >
            select.undefined
          </option>
          <option
            value="unreleased"
          >
            select.unreleased
          </option>
          <option
            value="deleted"
          >
            select.deleted
          </option>
        </select>
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="description"
        >
          values.description
        </label>
        <textarea
          aria-invalid={false}
          className="form-control"
          disabled={false}
          id="description"
          name="description"
          onChange={[Function]}
          placeholder="placeholder.description"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="contract"
        >
          values.contract
        </label>
        <div
          className=" css-b62m3t-container"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-2-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className="option-search__control css-13cymwt-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className="option-search__value-container css-1fdsijx-ValueContainer"
            >
              <div
                className="option-search__placeholder css-1jqq78o-placeholder"
                id="react-select-2-placeholder"
              >
                placeholder.contract
              </div>
              <div
                className="option-search__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-2-placeholder"
                  aria-expanded={false}
                  aria-haspopup={true}
                  aria-invalid={false}
                  aria-label="placeholder.contract"
                  aria-labelledby="contract"
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  className="option-search__input"
                  disabled={false}
                  id="contract"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  spellCheck="false"
                  style={
                    {
                      "background": 0,
                      "border": 0,
                      "color": "inherit",
                      "font": "inherit",
                      "gridArea": "1 / 2",
                      "label": "input",
                      "margin": 0,
                      "minWidth": "2px",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="option-search__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className="option-search__indicator-separator css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className="option-search__indicator option-search__dropdown-indicator css-1xc3v61-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <input
            name="contract"
            type="hidden"
            value=""
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="criticality"
        >
          values.criticality
        </label>
        <select
          aria-invalid={false}
          className="form-select"
          disabled={false}
          id="criticality"
          name="criticality"
          onChange={[Function]}
          placeholder="placeholder.criticality"
          value="normal"
        >
          <option
            value="critical"
          >
            select.critical
          </option>
          <option
            value="high"
          >
            select.high
          </option>
          <option
            value="normal"
          >
            select.normal
          </option>
          <option
            value="low"
          >
            select.low
          </option>
          <option
            value="noclass"
          >
            select.noclass
          </option>
        </select>
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="service_level"
        >
          values.service_level
        </label>
        <input
          aria-invalid={false}
          className="form-control"
          disabled={false}
          id="service_level"
          name="service_level"
          onChange={[Function]}
          placeholder="placeholder.service_level"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="service_type"
        >
          values.service_type
        </label>
        <select
          aria-invalid={false}
          className="form-select"
          disabled={false}
          id="service_type"
          name="service_type"
          onChange={[Function]}
          placeholder="placeholder.service_type"
          value="other"
        >
          <option
            value="onpremises"
          >
            select.onpremises
          </option>
          <option
            value="iaas"
          >
            select.iaas
          </option>
          <option
            value="paas"
          >
            select.paas
          </option>
          <option
            value="caas"
          >
            select.caas
          </option>
          <option
            value="faas"
          >
            select.faas
          </option>
          <option
            value="saas"
          >
            select.saas
          </option>
          <option
            value="other"
          >
            select.other
          </option>
        </select>
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="limitations"
        >
          values.limitations
        </label>
        <textarea
          aria-invalid={false}
          className="form-control"
          disabled={false}
          id="limitations"
          name="limitations"
          onChange={[Function]}
          placeholder="placeholder.limitations"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="related_services"
        >
          values.related_services
        </label>
        <div
          className=" css-b62m3t-container"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-3-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className="option-search__control css-13cymwt-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className="option-search__value-container option-search__value-container--is-multi css-1fdsijx-ValueContainer"
            >
              <div
                className="option-search__placeholder css-1jqq78o-placeholder"
                id="react-select-3-placeholder"
              >
                placeholder.related_services
              </div>
              <div
                className="option-search__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-3-placeholder"
                  aria-expanded={false}
                  aria-haspopup={true}
                  aria-invalid={false}
                  aria-label="placeholder.related_services"
                  aria-labelledby="related_services"
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  className="option-search__input"
                  disabled={false}
                  id="related_services"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  spellCheck="false"
                  style={
                    {
                      "background": 0,
                      "border": 0,
                      "color": "inherit",
                      "font": "inherit",
                      "gridArea": "1 / 2",
                      "label": "input",
                      "margin": 0,
                      "minWidth": "2px",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="option-search__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className="option-search__indicator-separator css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className="option-search__indicator option-search__dropdown-indicator css-1xc3v61-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div>
            <input
              name="related_services"
              type="hidden"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="required_installations"
        >
          values.required_installations
        </label>
        <div
          className=" css-b62m3t-container"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-4-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className="option-search__control css-13cymwt-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className="option-search__value-container option-search__value-container--is-multi css-1fdsijx-ValueContainer"
            >
              <div
                className="option-search__placeholder css-1jqq78o-placeholder"
                id="react-select-4-placeholder"
              >
                placeholder.required_installations
              </div>
              <div
                className="option-search__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-4-placeholder"
                  aria-expanded={false}
                  aria-haspopup={true}
                  aria-invalid={false}
                  aria-label="placeholder.required_installations"
                  aria-labelledby="required_installations"
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  className="option-search__input"
                  disabled={false}
                  id="required_installations"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  spellCheck="false"
                  style={
                    {
                      "background": 0,
                      "border": 0,
                      "color": "inherit",
                      "font": "inherit",
                      "gridArea": "1 / 2",
                      "label": "input",
                      "margin": 0,
                      "minWidth": "2px",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="option-search__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className="option-search__indicator-separator css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className="option-search__indicator option-search__dropdown-indicator css-1xc3v61-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div>
            <input
              name="required_installations"
              type="hidden"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <h2>
      service.responsibility
    </h2>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="product_owner"
        >
          values.product_owner
        </label>
        <input
          aria-invalid={false}
          className="form-control"
          disabled={false}
          id="product_owner"
          name="product_owner"
          onChange={[Function]}
          placeholder="placeholder.product_owner"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="service_holder"
        >
          values.service_holder
        </label>
        <input
          aria-invalid={false}
          className="form-control"
          disabled={false}
          id="service_holder"
          name="service_holder"
          onChange={[Function]}
          placeholder="placeholder.service_holder"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="provider"
        >
          values.provider
        </label>
        <div
          className=" css-b62m3t-container"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-5-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className="option-search__control css-13cymwt-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className="option-search__value-container css-1fdsijx-ValueContainer"
            >
              <div
                className="option-search__placeholder css-1jqq78o-placeholder"
                id="react-select-5-placeholder"
              >
                placeholder.provider
              </div>
              <div
                className="option-search__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-5-placeholder"
                  aria-expanded={false}
                  aria-haspopup={true}
                  aria-invalid={false}
                  aria-label="placeholder.provider"
                  aria-labelledby="provider"
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  className="option-search__input"
                  disabled={false}
                  id="provider"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  spellCheck="false"
                  style={
                    {
                      "background": 0,
                      "border": 0,
                      "color": "inherit",
                      "font": "inherit",
                      "gridArea": "1 / 2",
                      "label": "input",
                      "margin": 0,
                      "minWidth": "2px",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="option-search__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className="option-search__indicator-separator css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className="option-search__indicator option-search__dropdown-indicator css-1xc3v61-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <input
            name="provider"
            type="hidden"
            value=""
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="provider_role"
        >
          values.provider_role
        </label>
        <input
          aria-invalid={false}
          className="form-control"
          disabled={false}
          id="provider_role"
          name="provider_role"
          onChange={[Function]}
          placeholder="placeholder.provider_role"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="provider_contact"
        >
          values.provider_contact
        </label>
        <input
          aria-invalid={false}
          className="form-control"
          disabled={false}
          id="provider_contact"
          name="provider_contact"
          onChange={[Function]}
          placeholder="placeholder.provider_contact"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="additional_contacts"
        >
          values.additional_contacts
        </label>
        <textarea
          aria-invalid={false}
          className="form-control"
          disabled={false}
          id="additional_contacts"
          name="additional_contacts"
          onChange={[Function]}
          placeholder="placeholder.additional_contacts"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <h2>
      service.extra
    </h2>
  </div>,
  <div
    className="row"
  >
    <div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="customership"
        >
          values.customership
        </label>
        <div
          className=" css-b62m3t-container"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-6-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className="option-search__control css-13cymwt-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className="option-search__value-container option-search__value-container--is-multi css-1fdsijx-ValueContainer"
            >
              <div
                className="option-search__placeholder css-1jqq78o-placeholder"
                id="react-select-6-placeholder"
              >
                placeholder.customership
              </div>
              <input
                aria-autocomplete="list"
                aria-describedby="react-select-6-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                aria-invalid={false}
                aria-label="placeholder.customership"
                aria-labelledby="customership"
                aria-readonly={true}
                className="css-mohuvp-dummyInput-DummyInput"
                disabled={false}
                id="customership"
                inputMode="none"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                tabIndex={0}
                value=""
              />
            </div>
            <div
              className="option-search__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className="option-search__indicator-separator css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className="option-search__indicator option-search__dropdown-indicator css-1xc3v61-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div>
            <input
              name="customership"
              type="hidden"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="validity_type"
        >
          values.validity_type
        </label>
        <select
          aria-invalid={false}
          className="form-select"
          disabled={false}
          id="validity_type"
          name="validity_type"
          onChange={[Function]}
          placeholder="placeholder.validity_type"
          value="ongoing"
        >
          <option
            value="ongoing"
          >
            select.ongoing
          </option>
          <option
            value="temporary"
          >
            select.temporary
          </option>
        </select>
      </div>
    </div>
  </div>,
  <div>
    <button
      className="action-button btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      service.create
    </button>
  </div>,
]
`;
